ttnn/cpp/ttnn/operations/data_movement/split/split.cpp

```diff
@@ -71,9 +141,13 @@ std::vector<ttnn::Tensor> SplitOperation::invoke(
     const std::optional<MemoryConfig>& memory_config_arg) {
 
     auto memory_config = memory_config_arg.value_or(input_tensor.memory_config());
-    TT_FATAL(num_splits == 2, "Currently only supporting split in 2");
-    return detail::split_dim_two_chunks_tiled(input_tensor, dim, memory_config);
 
+    if (input_tensor.get_layout() == Layout::ROW_MAJOR) {
+        return detail::split_dim_n_chunks_rm(input_tensor, dim, num_splits,  memory_config);
+    } else {
+        TT_FATAL(num_splits == 2, "Currently only supporting split in 2");
```

Comment:
consider
```suggestion
        TT_FATAL(num_splits == 2, "ttnn.split currently only supports split in 2 in tiled layout, but {} is passed", num_splits);
```

Reply:
and maybe better move this check inside `split_dim_two_chunks_tiled`

Reply:
Yeah, agreed that we should move this.